src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
+
+  static removeDeadAnimal() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    super.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden && prey.health > 0) {
```

Comment:
[CODE PERFOMANCE] - There's no need to check if `prey.health > 0` inside the `if` statement, as the `Animal.removeDeadAnimal` method will handle filtering out animals with health less than or equal to 0.